internal/checker/services.go

```diff
@@ -618,3 +618,67 @@ func (c *Checker) GetTypeParameterAtPosition(s *Signature, pos int) *Type {
 	}
 	return t
 }
+
+func (c *Checker) GetContextualDeclarationsForObjectLiteralElement(objectLiteral *ast.Node, name string) []*ast.Node {
+	var result []*ast.Node
+	if t := c.getApparentTypeOfContextualType(objectLiteral, ContextFlagsNone); t != nil {
+		for _, t := range t.Distributed() {
+			prop := c.getPropertyOfType(t, name)
+			if prop != nil {
+				for _, declaration := range prop.Declarations {
+					result = core.AppendIfUnique(result, declaration)
+				}
+			} else {
+				for _, info := range c.getApplicableIndexInfos(t, c.getStringLiteralType(name)) {
+					if info.declaration != nil {
+						result = core.AppendIfUnique(result, info.declaration)
+					}
+				}
+			}
+		}
+	}
+	return result
+}
+
+var knownGenericTypeNames = map[string]struct{}{
+	"Array":            {},
+	"ArrayLike":        {},
+	"ReadonlyArray":    {},
+	"Promise":          {},
+	"PromiseLike":      {},
+	"Iterable":         {},
+	"IterableIterator": {},
+	"AsyncIterable":    {},
+	"Set":              {},
+	"WeakSet":          {},
+	"ReadonlySet":      {},
+	"Map":              {},
+	"WeakMap":          {},
+	"ReadonlyMap":      {},
+	"Partial":          {},
+	"Required":         {},
+	"Readonly":         {},
+	"Pick":             {},
+	"Omit":             {},
```

Comment:
```suggestion
	"Omit":             {},
	"Exclude":             {},
```

Reply:
However, we should include `NonNullable<T>`.